### DISCUSSION TITLE
Can I deploy PrivateGPT on a hosting platform, like a server or Cloud service?

### DISCUSSION THREAD

**Discussion by restartmedia** [imartinez/privateGPT#1530] [General]
Can I deploy PrivateGPT on a hosting platform, like a server or Cloud service so that my customers can access the cahtbot of privategpt through their browser?

If yes, then kindly advise which will be less expensive paltform to deploy and host the privategpt.

**Comment by lsmpascal** [comment]
May be i'm late.
Yes you can.
Check if python > 3.9 is installed.
```
python3 --version
```

Otherwise, install it.
```
sudo apt update && sudo apt upgrade
sudo apt install python3 -y
```
Then : 
```
# Git
sudo apt install git -y
git --version

# DL
git clone https://github.com/imartinez/privateGPT && cd privateGPT
# Env install
apt install python3.11-venv -y
# env creation
python3.11 -m venv .venv && source .venv/bin/activate
# Poetry install
pip install --upgrade pip poetry && poetry install --with ui,local && ./scripts/setup
# Open the fire wall
sudo ufw allow 8001
```

Then launch a browser and go to http://ip.of.the.server:8001
That's all.

The installation works perfectly, I tested it a lot of time on several servers.

On the "less expensive platform", for the moment I have no good answers.
I tested most of the <400$/month on vultr and the perf are terrible.
If you have a solution on this, please let me know !
